service/frontend/workflowHandler.go

```diff
@@ -3502,3 +3538,105 @@ func (wh *WorkflowHandler) checkNamespaceMatch(requestNamespace string, tokenNam
 func (wh *WorkflowHandler) metricsScope(ctx context.Context) metrics.Scope {
 	return interceptor.MetricsScope(ctx, wh.GetLogger())
 }
+
+func (wh *WorkflowHandler) makeFakeContinuedAsNewEvent(
+	ctx context.Context,
+	namespace string,
+	namespaceID string,
+	execution *commonpb.WorkflowExecution,
+	currentBranchToken []byte,
+	lastEvent *historypb.HistoryEvent,
+) (*historypb.HistoryEvent, error) {
+	switch lastEvent.EventType {
+	case enumspb.EVENT_TYPE_WORKFLOW_EXECUTION_COMPLETED:
+		if lastEvent.GetWorkflowExecutionCompletedEventAttributes().GetNewExecutionRunId() == "" {
+			return nil, nil
+		}
+	case enumspb.EVENT_TYPE_WORKFLOW_EXECUTION_FAILED:
+		if lastEvent.GetWorkflowExecutionFailedEventAttributes().GetNewExecutionRunId() == "" {
+			return nil, nil
+		}
+	case enumspb.EVENT_TYPE_WORKFLOW_EXECUTION_TIMED_OUT:
+		if lastEvent.GetWorkflowExecutionTimedOutEventAttributes().GetNewExecutionRunId() == "" {
+			return nil, nil
+		}
+	default:
+		return nil, nil
+	}
+
+	// We need to replace the last event. Get the run's start event so we can pull some fields out of it.
+	history, _, err := wh.getHistory(
```

Comment:
Do we really need to read the start event here? Can this information be retrieved from mutable state?

Reply:
It looks like `Input`, `LastCompletionResult`, and `Header` aren't available. Everything else appears to be. Do we need those? No, but really, we don't need to fill in anything at all except for `NewExecutionRunId`, since that's all the client will look at. We could leave everything else unset, though it might be a little confusing if anyone looks at once of these events.

Reply:
Considering this behavior triggers for a very targeted use case I would keep it as simple as possible.  If mutable state has everything we need to make it work then I would stay away from reading the first event to fill up those fields which are not even used by the logic we are trying make work.

Reply:
Sure, that makes sense.